Indicators/DeMarkerIndicator.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// In the DeMarker strategy, for some period of size N, set:
+    /// <para>
+    /// DeMax = High - Previous High, and 
+    /// DeMin = Previous Low - Low
+    /// </para>
+    /// where, in the prior, if either term is less than zero (DeMax or DeMin), set it to zero.
+    /// We can now define the indicator itself, DeM, as:
+    ///<para>
+    /// DeM = MA(DeMax)/(MA(DeMax)+MA(DeMin))
+    ///</para>
+    /// where MA denotes a Moving Average of period N.
+    /// 
+    /// https://www.investopedia.com/terms/d/demarkerindicator.asp
+    /// </summary>
+    public class DeMarkerIndicator : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private IndicatorBase<IndicatorDataPoint> _maxMA;
+        private IndicatorBase<IndicatorDataPoint> _minMA;
```

Comment:
These fields could be `readonly` since they are only defined once in the constructor.